src/traces/bar/plot.js

```diff
@@ -215,35 +215,179 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts, makeOnCompleteCallback)
                 (v > vc ? Math.ceil(v) : Math.floor(v));
             }
 
+            var op = Color.opacity(mc);
+            var fixpx = (op < 1 || lw > 0.01) ? roundWithLine : expandToVisible;
             if(!gd._context.staticPlot) {
                 // if bars are not fully opaque or they have a line
                 // around them, round to integer pixels, mainly for
                 // safari so we prevent overlaps from its expansive
                 // pixelation. if the bars ARE fully opaque and have
                 // no line, expand to a full pixel to make sure we
                 // can see them
-
-                var op = Color.opacity(mc);
-                var fixpx = (op < 1 || lw > 0.01) ? roundWithLine : expandToVisible;
-
                 x0 = fixpx(x0, x1, isHorizontal);
                 x1 = fixpx(x1, x0, isHorizontal);
                 y0 = fixpx(y0, y1, !isHorizontal);
                 y1 = fixpx(y1, y0, !isHorizontal);
             }
 
+            // Function to convert from size axis values to pixels
+            var c2p = isHorizontal ? xa.c2p : ya.c2p;
+
+            // Decide whether to use upper or lower bound of current bar stack
+            // as reference point for rounding
+            var outerBound;
+            if(di.s0 > 0) {
+                outerBound = di._sMax;
+            } else if(di.s0 < 0) {
+                outerBound = di._sMin;
+            } else {
+                outerBound = di.s1 > 0 ? di._sMax : di._sMin;
+            }
+
+            // Calculate corner radius of bar in pixels
+            function calcCornerRadius(radiusParam) {
+                var barWidth = isHorizontal ? Math.abs(y1 - y0) : Math.abs(x1 - x0);
+                var barLength = isHorizontal ? Math.abs(x1 - x0) : Math.abs(y1 - y0);
+                var stackedBarTotalLength = fixpx(Math.abs(c2p(outerBound, true) - c2p(0, true)));
+                var maxRadius = di.hasB ? Math.min(barWidth / 2, barLength / 2) : Math.min(barWidth / 2, stackedBarTotalLength);
+                var rPx;
+                if(!radiusParam) {
+                    return 0;
+                } else if(isNumeric(radiusParam)) {
+                    // If radius is already given as a number of pixels, use the given value
+                    rPx = radiusParam;
```

Comment:
We need to cast strings to number using + operator so that "10" becomes 10.
```suggestion
                    rPx = +radiusParam;
```